File: src/DutchMetar.Core/Features/Web/AirportDetails/Models/AirportDetails.cs
using DutchMetar.Core.Helpers.Enums;

namespace DutchMetar.Core.Features.Web.AirportDetails.Models;

public class AirportDetails
{
    public required string Icao { get; set; }
    
    public string? Name { get; set; }

    public MetarMeteoCondition MeteoCondition { get; set; } = MetarMeteoCondition.None;

    public DateTimeOffset LastUpdated { get; set; } = DateTimeOffset.MinValue;
    
    public AirportCurrentMetar? LatestWeather { get; set; }
    
    public AirportForecast? LatestForecast { get; set; }
}

File: src/DutchMetar.Web.Client/src/app/shared/models/airport/airport-details.ts
import { AirportCurrentMetar } from './airport-current-metar';
import { MeteoCondition } from '../../types/meteo-condition';
import { AirportForecast } from './airport-forecast';

export type AirportDetails = {
    icao: string;
    name?: string;
    meteoCondition: MeteoCondition;
    lastUpdated: Date;
    latestWeather?: AirportCurrentMetar
    latestForecast?: AirportForecast
}
